### PR TITLE
Messaging rework

### PR DESCRIPTION
**Pull request by Xaver106**
Changes the Messages so they will be parsed with [MiniMessage](https://docs.adventure.kyori.net/minimessage). Adds full customization for Messages.

### CODE REVIEW

**Comment by Xaver106** [comment]
This will not really work with the old language files! There is something I could do, but as it is not completely fixing old lang files, I don't think it is worth it.

**Comment by Xaver106** [comment]
@BigTallahasee Can't add you for review...

File: pom.xml [hunk 1]
```diff
@@ -24,20 +32,15 @@
 				<artifactId>maven-compiler-plugin</artifactId>
 				<version>3.8.1</version>
 				<configuration>
-					<source>1.7</source>
-					<target>1.7</target>
+					<source>17</source>
```
Comment (PyvesB) [inline pom.xml:35]: According to bstats.org, only ~60% of servers have migrated to Java 17 so far, it seems premature to switch to this version at this point.

On a similar note, the plugin supports all Spigot versions from 1.8 onwards. Versions 1.15 and previous still represent over 20% of marketshare. Older Spigot versions won't necessarily run on recent Java versions. I haven't checked, but there may be other breaking changes to that 1.8+ compatibility in this PR.

I'm not saying we should never drop support for old Java and Minecraft versions, but my personal gut feeling is that any change that abandons more than 10% of users should be thought about twice. That depends on the change of course, it's a balance to strike between amount of dropped users and benefits to the remaining ones.

@BigTallahasee you may want to add bstats.org to the plugin. The user share of Pet Master is probably different to that of global server statistics. :wink:

Comment (Xaver106) [reply pom.xml:35]: I can change it back, but as virtually nobody uses Java 7 anymore I would leave it with Java 8

**Review by BigTallahasee — COMMENTED** [review]
Looks good, If it would be easier you guys add me on discord BigTallahasee#2238. Here is also where I have been doing all the support and stuff from. https://discord.gg/PSZS7WnuG2

File: pom.xml (hunk 1 above)
Comment (BigTallahasee) [reply pom.xml]: > According to bstats.org, only ~60% of servers have migrated to Java 17 so far, it seems premature to switch to this version at this point.
> 
> On a similar note, the plugin supports all Spigot versions from 1.8 onwards. Versions 1.15 and previous still represent over 20% of marketshare. Older Spigot versions won't necessarily run on recent Java versions. I haven't checked, but there may be other breaking changes to that 1.8+ compatibility in this PR.
> 
> I'm not saying we should never drop support for old Java and Minecraft versions, but my personal gut feeling is that any change that abandons more than 10% of users should be thought about twice. That depends on the change of course, it's a balance to strike between amount of dropped users and benefits to the remaining ones.
> 
> @BigTallahasee you may want to add bstats.org to the plugin. The user share of Pet Master is probably different to that of global server statistics. 😉

Sounds good I will look into getting that added asap.

**Comment by Xaver106** [comment]
I recommend adding a warning about the breaking messages to the Update log, mostly all the colors and hovers break. Regenerating the default Lang.yml should be reccomended!

**Comment by BigTallahasee** [comment]
Ok sounds good I will get on there when I get home and will get that done

**Pull request merged**